qr-generator-app/lib/feature/display_qr/display_qr_page.dart

```diff
@@ -0,0 +1,129 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import 'package:qr_generator_app/feature/display_qr/display_qr_state_notifier.dart';
+import 'package:qr_generator_app/model/qr_data.dart';
+
+class DisplayQrPage extends StatelessWidget {
+  const DisplayQrPage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Display QR"),
+      ),
+      body: Consumer(
+        builder: (BuildContext context, WidgetRef ref, Widget? child) {
+          return Center(
+            child: ref.watch(displayQrStateNotifierProvider).when(
+                  initial: () => _buildDisplayQrButton(() => ref
+                      .read(displayQrStateNotifierProvider.notifier)
+                      .getSeed()),
+                  loading: () => Row(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      const SizedBox.square(
+                        dimension: 24.0,
+                      ),
+                      Padding(
+                        padding: const EdgeInsets.symmetric(horizontal: 16.0),
+                        child: _buildDisplayQrButton(null),
+                      ),
+                      const SizedBox.square(
+                        dimension: 24.0,
+                        child: CircularProgressIndicator(),
+                      ),
+                    ],
+                  ),
+                  success: (QrData qrData) => QrDataWidget(qrData: qrData),
+                  error: (String? error) => Row(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      const Text("an error occurred, press to retry."),
+                      IconButton(
+                          onPressed: () => ref
+                              .read(displayQrStateNotifierProvider.notifier)
+                              .getSeed(),
+                          icon: const Icon(Icons.restart_alt_rounded))
+                    ],
+                  ),
+                ),
+          );
+        },
+      ),
+    );
+  }
+
+  _buildDisplayQrButton(VoidCallback? action) => ElevatedButton.icon(
```

Comment:
**bug**: we never use functions to return widgets. You should really prefer widget over functions! There are a loot of benefits, most of all being able to use a `const` constructor and assigning keys. 

It's a good practice enforced by the Flutter team too ([see here](https://api.flutter.dev/flutter/widgets/StatelessWidget-class.html) the 'Performance consideration' part), you should really avoid doing this